### ISSUE TITLE
Parsing ISO8601 date with .000Z no longer works.

### ISSUE THREAD

**Issue by geoherna** [golang-module/carbon#155]
Hello,

I encountered an issue with parsing an ISO8601 timestamp when the ".000Z" is included. This used to work as expected in my previously used version **v2.0.1**, but this to not work anymore.

Consider the following code:
```go
carbon.Parse("2022-04-01T23:00:00.000Z").IsZero()
```

golang version: **1.18**

carbon version: **v2.1.9**

I expected to get:

```
false
```

But I actually get:

```
true
```
On a quick investigation, it seems that the removal of the following block might be the cause of the issue: https://github.com/golang-module/carbon/blob/f0fe61dd83557cfb27fc3a03e730a66d78b42f03/parser.go#L25
Noticed this only happened when upgraded to **v2.1.9**

Thanks!

**Comment by geoherna** [comment]
Going to close this issue — seems like I answered my own question with the example posted. The solution was to simply use `ParseByLayout` with `RFC3339Format` as the layout.

**Comment by gouguoyin** [comment]
Yes, at present, the Parse() method is too restrictive and not general enough, and is being optimized

**Comment by gouguoyin** [comment]
It has been fixed in 1.7.0 or 2.2.0
